src/run/cluster.cpp

```diff
@@ -151,10 +244,273 @@ void run() {
 		}
 	}
 
-	db->close();
 	if (out != &cout) delete out;
 	delete rep_seqs;
 	delete rep_ids;
 }
 
-}}
\ No newline at end of file
+
+void run_transitive_closure_clustering(DatabaseFile& db){
+	// testing
+	// LazyDisjointIntegralSet<uint32_t> test(5);
+	// test.merge(1,3);
+	// test.merge(3,4);
+	// test.merge(0,2);
+	// auto los = test.getListOfSets();
+	// for(auto& set: los){
+	// 	for(auto item:set){
+	// 		cout<< " " << item;
+	// 	}
+	// 	cout<<endl;
+	// }
+	// unordered_set<string> s ( {"red","green","blue","rain", "sun", "snow"} );
+	// LazyDisjointTypeSet<string> ts(&s);
+
+	// ts.merge("red", "green");
+	// ts.merge("snow", "rain");
+	// ts.merge("sun", "rain");
+	// auto lo = ts.getListOfSets();
+	// for(auto& set: lo){
+	// 	cout<< "set :";
+	// 	for(auto item:set){
+	// 		cout<< " " << item;
+	// 	}
+	// 	cout<<endl;
+	// }
+
+	const size_t seq_count = db.ref_header.sequences;
+	statistics.reset();
+	config.command = Config::blastp;
+	config.no_self_hits = true;
```

Comment:
The `no_self_hits` option probably needs to be reworked for self-alignment, as it relies on accession and alignment length, which could be broken by repeat masking (see align/culling.cpp:90).

Reply:
So it looks like setting it to false would be the preferred option here anyways. I think I just kept it because I was unsure as the numbers of clusters were so different (but we have resolved that in our email conversation). For MCL it is preferable to have the self-hits included as the graph needs to have these self-loops (and which I am currently adding). This will also enable more general similarity measures to be used for MCL (I don't like the idea of adding a "neutral" value as suggested in one paper).
At work, I have observed though that sometimes diamond does not find self-hits even if this option is enabled, but I don't know if that was a bug in the specific version of diamond we have been running for a while (and of course I have not checked since). I will enable this and throw a warning if a diagonal element is missing in the matrix.

Reply:
Not finding a self hit should be either due to the repeat masking or due to the hit not being below the evalue cutoff. I'm not aware of any other way this could happen, but let me know if you find an example.

Reply:
Will do. Resolving this conversation for now.